master/buildbot/test/unit/test_reporter_http.py

```diff
@@ -70,12 +70,18 @@ def setUp(self):
         yield self.master.startService()
 
     @defer.inlineCallbacks
-    def createReporter(self, **kwargs):
+    def createReporter(self, auth=("username", "passwd"), **kwargs):
         self._http = yield fakehttpclientservice.HTTPClientService.getService(
             self.master,
-            "serv", auth=("username", "passwd"))
-        self.sp = sp = HttpStatusPush("serv", auth=(
-            "username", Interpolate("passwd")), **kwargs)
+            "serv", auth=auth)
+
+        interpolated_auth = None
+        if auth is not None:
+            username, passwd = auth
+            passwd = Interpolate(passwd)
```

Comment:
Interpolate is immutable, so you can put it in default args

Reply:
Non-interpolated passwd was used in getService so I couldn't reuse the interpolated value.